src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -37,6 +37,10 @@
 
 logger = logging.getLogger(__name__)
 
+# Special (fake) exit code for jobs that are lost due to preemption.
+# We use a code that is outside of the normal 0-255 range on purpose here to ensure it
+# doesn't conflict with other exit codes.
+LOST_JOB_EXIT_STATUS = 2550
```

Comment:
I still think this should be a non-integer, and we should go through and change things to understand that the return value isn't always an exit code. Right now the Kubernetes batch system spits out -1 in cases where it couldn't get an exit code; we could also change that to something more descriptive.

I think to make it work all we would really have to change is https://github.com/DataBiosphere/toil/blob/8e1d8c0f45f6210bbd0900380b62c5a1c1a6cdc7/src/toil/leader.py#L498

Even if we leave it like this, we shouldn't call this an EXIT_STATUS; it should be something else (stop reason?) and the getUpdatedBatchJob docs should be changed accordingly.

Reply:
I still feel uneasy about changing the type to be `stri|int`: while we can ensure the code in the current state can handle that, any future implementation requires thinking about both cases separately and may introduce bugs.

What do you think about adding another parameter to the tuple, `exitReason: str|None`, that's independent of `exitStatus`? The new field could be none for majority of cases, but we could have special ones for cases that we care about.

Doing this properly would require refactoring all use cases to (ideally) use a class/namedtuple [*], so that the code is more readable, but I think it's worth the effort and we could add additional fields easier in the future.

[*] Using namedtuple would be ideal, but defaults are not easily supported before python3.7. We could require all setters to set all fields, but it would make it more difficult if we wanted to add new fields later.